metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/WorkflowExecutionDao.java

```diff
@@ -544,11 +540,11 @@ private Filter createFilter(Set<String> datasetIds, Set<PluginStatus> pluginStat
         .elemMatch(METIS_PLUGINS.getFieldName(), elemMatchFilters.toArray(Filter[]::new));
 
     final Filter collectedFilters;
-    if (datasetIds != null) {
+    if (CollectionUtils.isEmpty(datasetIds)) {
```

Comment:
I'm looking now at `OrchestratorService.getDatasetIdsToFilterOn()`. This returns null if the user is admin, and a non-null list if the user is not admin. So hypothetically, if a user is not admin but has no datasets in their organization, the workflow list should be empty. But with this condition, it is actually now returning all datasets (i.e. the user is treated as admin). Is this not a problem?

Reply:
Good point I will revert this and provide a comment for it

Reply:
Actually we can even avoid sending a request with the list is empty